sled-agent/src/rack_setup/service.rs

```diff
@@ -1307,4 +1287,166 @@ impl ServiceInner {
+        info!(self.log, "Building Uplink Configuration");
+        let ipv6_entry = Ipv6Entry {
+            addr: BOUNDARY_SERVICES_ADDR.parse().map_err(|e| {
+                SetupServiceError::BadConfig(format!(
+                "failed to parse `BOUNDARY_SERVICES_ADDR` as `Ipv6Addr`: {e}"
+            ))
+            })?,
+            tag: "rss".into(),
+        };
+        let mut dpd_port_settings = PortSettings {
+            tag: "rss".into(),
+            links: HashMap::new(),
+            v4_routes: HashMap::new(),
+            v6_routes: HashMap::new(),
+        };
+        let link_id = LinkId(0);
+        let addr = IpAddr::V4(uplink_config.uplink_ip);
+        let link_settings = LinkSettings {
+            // TODO Allow user to configure link properties
+            // https://github.com/oxidecomputer/omicron/issues/3061
+            params: LinkCreate {
+                autoneg: false,
+                kr: false,
+                fec: convert_fec(&uplink_config.uplink_port_fec),
+                speed: convert_speed(&uplink_config.uplink_port_speed),
+            },
+            addrs: vec![addr],
+        };
+        dpd_port_settings.links.insert(link_id.to_string(), link_settings);
+        let port_id: PortId = uplink_config
+            .uplink_port
+            .parse()
+            .map_err(|e| SetupServiceError::BadConfig(
+            format!("could not use value provided to rack_network_config.uplink_port as PortID: {e}")))?;
+        let nexthop = Some(uplink_config.gateway_ip);
+        dpd_port_settings.v4_routes.insert(
+            Ipv4Cidr { prefix: "0.0.0.0".parse().unwrap(), prefix_len: 0 }
+                .to_string(),
+            RouteSettingsV4 {
+                link_id: link_id.0,
+                vid: uplink_config.uplink_vid,
+                nexthop,
+            },
+        );
+        Ok((ipv6_entry, dpd_port_settings, port_id))
+    }
+
+    async fn wait_for_dendrite(&self, dpd: &DpdClient) {
+        loop {
+            info!(self.log, "Checking dendrite uptime");
+            match dpd.dpd_uptime().await {
+                Ok(uptime) => {
+                    info!(self.log, "Dendrite online"; "uptime" => uptime.to_string());
+                    break;
+                }
+                Err(e) => {
+                    info!(self.log, "Unable to check Dendrite uptime"; "reason" => format!("{e}"));
+                }
+            }
+            info!(self.log, "Waiting for dendrite to come online");
+            tokio::time::sleep(std::time::Duration::from_secs(2)).await;
+        }
+    }
+}
+/// Scan the DNS configuration in the service plan to find
+/// the addresses of the switch zones
+fn switch_zone_addresses(service_plan: &ServicePlan) -> Vec<Ipv6Addr> {
+    let mut addrs = vec![];
+    for entry in &service_plan.dns_config.zones {
+        for (name, records) in &entry.records {
+            if name.starts_with("dendrite") {
```

Comment:
Rather than scan through the service plan and string compare here can we instead add it as a top-level value to `ServicePlan`

Reply:
I _think_ after rebasing I can now fetch all Dendrite addresses from DNS, making this logic unnecessary. Will test this and update accordingly.

Reply:
Initial testing shows this works. I need to test in a dual-scrimlet configuration to be sure, will update.

Reply:
I was able to test to the limits of what my current lab hardware will support, all seems well with this change. :)